### PR TITLE
[nfc][PGO]Factor out profile scaling into a standalone helper function

### PR DESCRIPTION
**Pull request by mingmingl-llvm**
- Put the helper function in `ProfDataUtil.h/cpp`, which is already a dependency of `Instructions.cpp`
- The helper function could be re-used to update profiles of `InvokeInst` (in a follow-up pull request)

### CODE REVIEW

File: llvm/test/Transforms/Inline/update_invoke_value_profile.ll [hunk 1]
```diff
@@ -0,0 +1,54 @@
+; A pre-commit test to show that value profiles associated with invoke are not updated.
```
Comment (david-xl) [inline llvm/test/Transforms/Inline/update_invoke_value_profile.ll:1]: Can the two test cases be merged into one? i.e.,  callee makes calls to inner_callee and *func .

File: llvm/test/Transforms/Inline/update_value_profile.ll [hunk 2]
```diff
@@ -0,0 +1,82 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 4
```
Comment (david-xl) [inline llvm/test/Transforms/Inline/update_value_profile.ll:1]: remove this line?

File: llvm/test/Transforms/Inline/update_value_profile.ll [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+!12 = !{i32 10000, i64 100, i32 1}
+!13 = !{i32 999000, i64 100, i32 1}
+!14 = !{i32 999999, i64 1, i32 2}
+!17 = !{!"VP", i32 0, i64 1600, i64 15186643663281959480, i64 1000, i64 15101948577241817854, i64 600}
+!18 = !{!"function_entry_count", i64 1000}
+!19 = !{!"function_entry_count", i64 600}
+!20 = !{!"function_entry_count", i64 1700}
+;.
+; CHECK: [[PROF0]] = !{!"function_entry_count", i64 100}
+; CHECK: [[PROF1]] = !{!"VP", i32 0, i64 94, i64 -3260100410427592136, i64 58, i64 -3344795496467733762, i64 35}
```
Comment (david-xl) [inline llvm/test/Transforms/Inline/update_value_profile.ll:77]: why does the md5sum change (compared with !17)?

**Review by david-xl — APPROVED** [review]

File: llvm/test/Transforms/Inline/update_value_profile.ll (hunk 2 above)
Comment (minglotus-6) [reply llvm/test/Transforms/Inline/update_value_profile.ll]: done.

File: llvm/test/Transforms/Inline/update_value_profile.ll (hunk 3 above)
Comment (minglotus-6) [reply llvm/test/Transforms/Inline/update_value_profile.ll]: Ah I didn't realize this; test passes (and the two numbers have the same hex representation).

I updated both to simpler values.

File: llvm/test/Transforms/Inline/update_invoke_value_profile.ll (hunk 1 above)
Comment (minglotus-6) [reply llvm/test/Transforms/Inline/update_invoke_value_profile.ll]: done.

**Review by david-xl — APPROVED** [review]

**Pull request merged**